docs/Additional_Build_Documentation.md

```diff
@@ -0,0 +1,296 @@
+the tests we need to include googletest, which can be done either by cloning the googletest
+repository or by asking benchmark in the configuration step to download this dependency.
+We can clone the googletest repository in benchmark using
+```bash
+$ git clone https://github.com/google/googletest.git benchmark/googletest
+```
+To configure Benchmark we use CMake as follows:
+```bash
+$ cmake -G <generator> [options] -S benchmark -B build
+```
+If we haven't cloned the googletest repository as described above we can ask benchmark to download this dependency in the above configuration step by adding `-DBENCHMARK_DOWNLOAD_DEPENDENCIES=ON`
+
+Then to build Benchmark we use CMake again as follows:
+```bash
+$ cmake -build build
+```
+If we don't want the Benchmark tests to build and run we can add `-DBENCHMARK_ENABLE_GTEST_TESTS=OFF` in the configuration step.
+
+## Using Benchmark
+
+If we want to use Benchmark with a particular file, say `BenchmarkEverything.cpp`,
+we can build Benchmark and use the following comands in the terminal.
+
+### In Windows
+```bash
+> cl /EHsc /MD BenchmarkEverything.cpp /I<BENCHMARK_SOURCE_DIR>\include /link shlwapi.lib <BENCHMARK_BUILD_DIR>\src\Release\benchmark.lib
+```
+where `<BENCHMARK_SOURCE_DIR>` is the directory where Benchmark was cloned into and `<BENCHMARK_BUILD_DIR>` the directory where Benchmark was build. Here, we had build Benchmark in the Release configuration and hence we 
+used `<BENCHMARK_BUILD_DIR>\src\Release\benchmark.lib`. We also used the `cl` compiler which is provided with Visual Studio and also linked the `shlwapi` library.
+
+### In Linux
+```bash
+$ g++ -std=c++11 BenchmarkEverything.cpp -I<BENCHMARK_SOURCE_DIR>/include -pthread -L <BENCHMARK_BUILD_DIR>\src -lbenchmark
+```
+where `<BENCHMARK_SOURCE_DIR>` is the directory where Benchmark was cloned into and `<BENCHMARK_BUILD_DIR>` the directory where Benchmark was build. We also need to link the `pthread` library.
+
+# Installing Benchmark
+
+After building Benchmark we can choose to install it in our systems. 
+
+For Ubuntu and Debian based systems we have to build Benchmark in the Release configuration by applying `-DCMAKE_BUILD_TYPE=RELEASE` in the configuration step. Then install it using `sudo make install` inside the build directory.
+
+We could also use Microsoft's C++ Library Manager [`vcpkg`](https://github.com/Microsoft/vcpkg) to install Benchmark in our systems. This manager works in Windows, Linux and Mac OS and it is very easy to manage libraries with it. To install Benchmark we simply need to run the following commnd, provided we already have `vcpkg` installed in our system with user-wide integration. (NOTE: documentation for installation of `vcpkg` can be found [here](https://github.com/Microsoft/vcpkg))
+
+For Windows
+```bash
+PS> .\vcpkg install benchmark
+```
+For Linux
+```bash
+$ ./vcpkg install benchmark
+```
+
+# Embedding Benchmark in other projects
```

Comment:
as above, this should come first _or_ be a different doc.

Reply:
embedding and using is in the file `docs/Using_Benchmark.md` and the installation using `vcpkg` has been moved to `README.md`